src/DatePicker.js

```diff
@@ -1,41 +1,28 @@
-import React from 'react';
+import React, { useMemo, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import TextInput from './TextInput';
 import idgen from './idgen';
 
-class DatePicker extends React.Component {
-  constructor(props) {
-    super(props);
+const DatePicker = props => {
+  const id = useMemo(() => props.id || `datepicker${idgen()}`, [props.id]);
 
-    this.id = props.id || `datepicker${idgen()}`;
-  }
-
-  componentDidMount() {
-    if (typeof M !== 'undefined') {
-      const { onChange } = this.props;
-      const elem = document.getElementById(this.id);
-      const options = onChange
-        ? { ...this.props.options, onSelect: onChange }
-        : this.props.options;
-
-      this.instance = M.Datepicker.init(elem, options);
-    }
-  }
+  useEffect(() => {
+    const { onChange } = props;
```

Comment:
let's deconstruct all required props in the component creation: `const Foo = ({ bar, baz }) =>`
and then we can be more specific in the hooks dependencies :)